adoc/admin-security-certificates.adoc

```diff
@@ -470,6 +483,272 @@ its `pki` folder. This file is a symlink to the latest generated certificate.
++
+[source,bash]
+----
+cat << EOF > issuer.yaml
+apiVersion: cert-manager.io/v1alpha3
+kind: Issuer
+metadata:
+  name: kubernetes-ca // <1>
+  namespace: kube-system
+spec:
+  ca:
+    secretName: kubernetes-ca // <2>
+EOF
+
+kubectl apply -f issuer.yaml
+----
+<1> The issuer name.
+<2> The secret reference name.
++
+[NOTE]
+====
+If you want to use custom trusted CA certificate/key to sign the certificate, create a custom trust CA issuer.
+
+[source,bash]
+----
+cat << EOF > custom-trusted-ca-issuer.yaml
+apiVersion: cert-manager.io/v1alpha3
+kind: Issuer // <1>
+metadata:
+  name: custom-trusted-ca
+  namespace: kube-system
+spec:
+  ca:
+    secretName: custom-trusted-ca
+EOF
+
+kubectl apply -f custom-trusted-ca-issuer.yaml
+----
+<1> Issuer or ClusterIssuer.
+====
+
+* Cert-Manager Certificate Resource
++
+The cert-manager has the concept of Certificates that define a desired x509 certificate which will be renewed and kept up to date.
++
+** oidc-dex certificate
++
+[source,bash]
+----
+cat << EOF > oidc-dex-certificate.yaml
+apiVersion: cert-manager.io/v1alpha3
+kind: Certificate
+metadata:
+  name: oidc-dex-cert
+  namespace: kube-system
+spec:
+  subject:
+    organizations:
+    - system:masters
+  commonName: oidc-dex
+  duration: 8760h # 1 year // <1>
+  renewBefore: 720h # 1 month // <2>
+  # At least one of a DNS Name or IP address is required.
+  dnsNames:
+  - $(cat admin.conf | grep server | awk '{print $2}' | sed 's/https:\/\///g' | sed 's/:6443//g') // <3>
+  ipAddresses:
+  - $(cat admin.conf | grep server | awk '{print $2}' | sed 's/https:\/\///g' | sed 's/:6443//g') // <4>
+  secretName: oidc-dex-cert
+  issuerRef:
+    name: kubernetes-ca // <5>
+    kind: Issuer // <6>
+    group: cert-manager.io
+  isCA: false
+  usages:
+    - digital signature
+    - key encipherment
+    - server auth
+  keySize: 2048
+  keyAlgorithm: rsa
+  keyEncoding: pkcs1
+EOF
+
+kubectl apply -f oidc-dex-certificate.yaml
+----
+<1> Certificate default duration, format (XhYmZs).
+<2> Certificate renew before expiration duration, format (XhYmZs).
+<3> DNSNames is a list of subject alt names to be used on the Certificate.
+<4> IPAddresses is a list of IP addresses to be used on the Certificate.
+<5> The cert-manager issuer name.
+<6> Issuer or ClusterIssuer.
++
+This certificate will tell cert-manager to attempt to use the Issuer named kubernetes-ca to obtain a certificate key pair for the domains lists in dnsNames and ipAddresses. If successful, the resulting key and certificate will be stored in a secret named oidc-dex-cert with keys of tls.key and tls.crt respectively. 
++
+The dnsNames and ipAddressed field specifies a list of Subject Alternative Names to be associated with the certificate.
++
+The referenced Issuer must exist in the same namespace as the Certificate. A Certificate can alternatively reference a ClusterIssuer which is non-namespaced and so can be referenced from any namespace.
```

Comment:
```suggestion
The referenced Issuer must exist in the same namespace as the Certificate. A Certificate can alternatively reference a ClusterIssuer which is cluster-wide so it can be referenced from any namespace.
```